store/gcworker/gc_worker.go

```diff
@@ -309,12 +309,29 @@ func (w *GCWorker) leaderTick(ctx context.Context) error {
 		return nil
 	}
 
+	concurrency, err := w.getGCConcurrency(ctx)
+	if err != nil {
+		logutil.Logger(ctx).Info("[gc worker] failed to get gc concurrency.",
+			zap.String("uuid", w.uuid),
+			zap.Error(err))
+		return errors.Trace(err)
+	}
+
 	ok, safePoint, err := w.prepare(ctx)
-	if err != nil || !ok {
-		if err != nil {
-			metrics.GCJobFailureCounter.WithLabelValues("prepare").Inc()
-		}
+	if err != nil {
+		metrics.GCJobFailureCounter.WithLabelValues("prepare").Inc()
 		return errors.Trace(err)
+	} else if !ok {
+		if w.logBackupEnabled {
+			tryResolveLocksTS, err := w.getTryResolveLocksTS()
+			if err != nil {
+				return errors.Trace(err)
+			}
+			if err = w.legacyResolveLocks(ctx, 0, tryResolveLocksTS, concurrency); err != nil {
```

Comment:
you can add some comments about why we need these steps and why we can pass 0 here